### DISCUSSION TITLE
add functionality similar to fastapi.APIRouter.add_api_route

### DISCUSSION THREAD

**Discussion by jeffective** [zauberzeug/nicegui#1534] [Ideas / Feature Requests]
Add a function to nicegui.APIRouter() that allows page declaration without using a decorator.

function could be something like nicegui.APIRouter.add_page_route

reason being this allows a page function to be declared within a class, and the page function would have access to self.

[reference](https://stackoverflow.com/questions/63853813/how-to-create-routes-with-fastapi-within-a-class)

**Comment by rodja** [comment]
Sounds good. Anyone interested in creating a pull request?
